packages/evm/contracts/BoostCore.sol

```diff
@@ -134,17 +134,33 @@ contract BoostCore is Ownable, ReentrancyGuard {
     function claimIncentive(uint256 boostId_, uint256 incentiveId_, address referrer_, bytes calldata data_)
         external
         payable
-        nonReentrant
     {
+        claimIncentive(boostId_, incentiveId_, referrer_, data_, msg.sender);
+    }
+
+    /// @notice Claim an incentive for a Boost on behalf of another user
+    /// @param boostId_ The ID of the Boost
+    /// @param incentiveId_ The ID of the Incentive
+    /// @param referrer_ The address of the referrer (if any)
+    /// @param data_ The data for the claim
+    /// @param claimant the address of the user eligible for the incentive payout
+    function claimIncentive(
```

Comment:
Upstream tooling get's confused when there's two functions with the same name and overloaded params - could we do like `claimIncentiveFor` or something along those lines. Preference for minimal extra characters